src/components/Identity/createComponent.js

```diff
@@ -17,135 +17,135 @@ const addEcidToPayload = (payload, ecid) => {
   });
 };
 
-export default (processIdSyncs, config, logger, optIn, eventManager) => {
+export default (processIdSyncs, config, logger, consent, eventManager) => {
   const { idMigrationEnabled, orgId } = config;
   const identityCookieName = getNamespacedCookieName(
     orgId,
     IDENTITY_COOKIE_KEY
   );
   let deferredForEcid;
-  const migration = createMigration(orgId);
+  const migration = createMigration({ orgId, consent });
   const hasIdentityCookie = () => Boolean(cookieJar.get(identityCookieName));
-  const customerIds = createCustomerIds(eventManager);
+  const customerIds = createCustomerIds({ eventManager, consent });
+
+  // TO-DOCUMENT: We wait for ECID before trigger any events.
+  const accommodateIdentityOnRequest = payload => {
+    let promise;
+
+    if (!hasIdentityCookie()) {
+      const ecidToMigrate =
+        idMigrationEnabled && migration.getEcidFromLegacyCookies();
```

Comment:
We deemed it okay to read the legacy cookie without consent, but we can't write it without consent.